packages/component-library/src/components/form/mt-link/mt-link.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+  <component
+    :is="as"
+    class="mt-link"
+    :class="linkClasses"
+    :aria-disabled="disabled"
+    :tabindex="disabled ? -1 : 0"
+    v-bind="to ? { ...$attrs, to } : $attrs"
+    @click="onClick"
+  >
+    <slot />
+  </component>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import type { PropType } from "vue";
+
+export default defineComponent({
+  name: "MtLink",
+
+  props: {
+    /**
+     * Set the element type of the link
+     */
+    as: {
+      type: String,
+      required: false,
+      default: "router-link",
+    },
+
+    /**
+     * Set the to prop of the router/nuxt-link
+     */
+    to: {
+      type: String,
+      required: false,
+      default: undefined,
+    },
+
+    /**
+     * Render the link in various styles
+     */
+    variant: {
+      type: String as PropType<"primary" | "critical">,
+      required: false,
+      default: "primary",
+      validator(value: string) {
+        if (!value.length) {
+          return true;
+        }
+        return ["primary", "critical"].includes(value);
+      },
+    },
+
+    /**
+     * Make the link unclickable
+     */
+    disabled: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  computed: {
+    linkClasses() {
+      return {
+        [`mt-link--${this.variant}`]: !!this.variant,
+        "mt-link--disabled": this.disabled,
+      };
+    },
+  },
+
+  methods: {
+    onClick(event: MouseEvent) {
+      if (this.disabled) {
+        event.preventDefault();
+        event.stopImmediatePropagation();
+        return;
+      }
+
+      this.$emit("click", event);
+    },
+  },
+});
+</script>
+
+<style lang="scss">
```

Comment:
It makes sense to scope these styles. This way other components cannot mess with the styles. It's still possible to apply margin to the root element so that you're able to position it in case you need that.

Reply:
Thank for the feedback @Haberkamp, I updated it.